compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1126,9 +1127,32 @@ static SmallVector<int64_t> getLinalgExtDefaultWorkgroupTileSizes(
   return workgroupTileSizes;
 }
 
+static LogicalResult setTransformStrategyRootConfig(func::FuncOp entryPointFn,
+                                                    tensor::PackOp op) {
+  if (!clCPUEnableTransformDialectJit) return failure();
+
+  // TODO(hanchung): Move the checks to matchAndSetPackStrategy. We can't move
+  // them to the method because of cyclic deps.
+  auto targetAttr = IREE::HAL::ExecutableTargetAttr::lookup(entryPointFn);
+  if (!isX86_64(targetAttr)) return failure();
+
+  cpu::PackConfig packConfig{.lowerToAVX2 = hasAVX2Feature(targetAttr)};
```

Comment:
Nice!

Can we refactor this a bit to start populating a single HW-related "model struct" that can be reused by all strategies ? 

This does not need to contain everything from day one but it needs to be separable from the notion of a "pack". The pack config should take such a object and make a decision based on a relevant subset of the available information (see links I posted in other comments).